# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/test/webgl-conf/checkout/conformance/context/context-lost-restored.html
# Commit: 8e0aea33f622
# Full Hash: 8e0aea33f622a76c5c8baa7dcf4485ffefd1662f
# Author: Kelsey Gilbert <kelsey.gilbert@mozilla.com>
# Date: 2024-07-09 09:45:28
# Regressor Bug: 1892542
# File Overlap Count: 1
# Description:
#   Bug 1892542 - Webgl e.g. createBuffer() made infallible. r=gfx-reviewers,webidl,saschanaz,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D208050
# ==============================================================================

diff -r b953e0ccb68b -r 8e0aea33f622 dom/canvas/test/webgl-conf/checkout/conformance/context/context-lost-restored.html
--- a/dom/canvas/test/webgl-conf/checkout/conformance/context/context-lost-restored.html	Mon Jul 08 19:36:12 2024 +0300
+++ b/dom/canvas/test/webgl-conf/checkout/conformance/context/context-lost-restored.html	Mon Jul 08 17:24:12 2024 +0000
@@ -239,13 +239,24 @@
 function testOESVertexArrayObject() {
   if (OES_vertex_array_object) {
     // Extension must still be lost.
-    shouldBeNull("OES_vertex_array_object.createVertexArrayOES()");
+    let vao = OES_vertex_array_object.createVertexArrayOES();
+    assertMsg(vao, "[with extension lost] createVertexArrayOES() -> non-null");
+    assertMsg(!OES_vertex_array_object.isVertexArrayOES(vao), "[with context lost] isVertexArrayOES(createVertexArrayOES()) -> false");
     // Try re-enabling extension
 
     old_OES_vertex_array_object = OES_vertex_array_object;
     OES_vertex_array_object = reGetExtensionAndTestForProperty(gl, "OES_vertex_array_object", false);
-    shouldBeTrue("OES_vertex_array_object.createVertexArrayOES() != null");
-    shouldBeTrue("old_OES_vertex_array_object.createVertexArrayOES() == null");
+
+    vao = OES_vertex_array_object.createVertexArrayOES();
+    assertMsg(vao, "[with new non-lost extension] createVertexArrayOES() -> non-null");
+    assertMsg(!OES_vertex_array_object.isVertexArrayOES(vao), "[with new non-lost extension, before bindVAO] isVertexArrayOES(createVertexArrayOES()) -> false");
+    OES_vertex_array_object.bindVertexArrayOES(vao);
+    OES_vertex_array_object.bindVertexArrayOES(null);
+    assertMsg(OES_vertex_array_object.isVertexArrayOES(vao), "[with new non-lost extension, after bindVAO] isVertexArrayOES(createVertexArrayOES()) -> true");
+
+    vao = old_OES_vertex_array_object.createVertexArrayOES();
+    assertMsg(vao, "[with old lost extension] createVertexArrayOES() -> non-null");
+    assertMsg(!old_OES_vertex_array_object.isVertexArrayOES(vao), "[with old lost extension] isVertexArrayOES(createVertexArrayOES()) -> false");
   }
 }
 